diff --git a/source/frontend/.storybook/baseMocks.ts b/source/frontend/.storybook/baseMocks.ts
index 367fb6f36..f3cb662d8 100644
--- a/source/frontend/.storybook/baseMocks.ts
+++ b/source/frontend/.storybook/baseMocks.ts
@@ -110,6 +110,14 @@ export const baseMocks = [
       items: [],
     })
   ),
+  http.get('http://localhost:4466/api/v1/pods', () =>
+    HttpResponse.json({
+      kind: 'PodList',
+      apiVersion: 'v1',
+      metadata: {},
+      items: [],
+    })
+  ),
   http.get('http://localhost:4466/apis/apiextensions.k8s.io/v1/customresourcedefinitions', () =>
     HttpResponse.json({
       kind: 'CustomResourceDefinitionList',
diff --git a/source/plugins/headlamp-plugin/config/.storybook/baseMocks.ts b/source/plugins/headlamp-plugin/config/.storybook/baseMocks.ts
index 5d0ae098b..6d3f13776 100644
--- a/source/plugins/headlamp-plugin/config/.storybook/baseMocks.ts
+++ b/source/plugins/headlamp-plugin/config/.storybook/baseMocks.ts
@@ -183,6 +183,14 @@ export const baseMocks = [
       items: [],
     })
   ),
+  http.get('http://localhost:4466/api/v1/pods', () =>
+    HttpResponse.json({
+      kind: 'PodList',
+      apiVersion: 'v1',
+      metadata: {},
+      items: [],
+    })
+  ),
   http.get('http://localhost:4466/apis/apiextensions.k8s.io/v1/customresourcedefinitions', () =>
     HttpResponse.json({
       kind: 'CustomResourceDefinitionList',
