diff --git a/source/frontend/src/components/resourceMap/GraphView.stories.tsx b/source/frontend/src/components/resourceMap/GraphView.stories.tsx
index 7159ad3fe..4fcd53bef 100644
--- a/source/frontend/src/components/resourceMap/GraphView.stories.tsx
+++ b/source/frontend/src/components/resourceMap/GraphView.stories.tsx
@@ -212,7 +212,7 @@ const mockSource: GraphSource = {
 
 export const BasicExample = () => (
   <TestContext>
-    <GraphView height="600px" defaultSources={[mockSource]} />
+    <GraphView height="600px" defaultSources={[mockSource]} defaultRelations={[]} />
   </TestContext>
 );
 BasicExample.args = {};
