diff --git a/source/app/electron/main.ts b/source/app/electron/main.ts
index a2ac8e598..2ffbbe631 100644
--- a/source/app/electron/main.ts
+++ b/source/app/electron/main.ts
@@ -17,2 +17,2 @@
-import { ChildProcessWithoutNullStreams, exec, execSync, spawn } from 'child_process';
+import { ChildProcessWithoutNullStreams, exec, execFileSync, execSync, spawn } from 'child_process';
 import { randomBytes } from 'crypto';
@@ -134,3 +134,3 @@ const args = yargs(hideBin(process.argv))
         const backendPath = path.join(process.resourcesPath, 'headlamp-server');
-        const stdout = execSync(`${backendPath} list-plugins`);
+        const stdout = execFileSync(backendPath, ['list-plugins']);
         process.stdout.write(stdout);
