diff --git a/source/frontend/src/components/common/Table/Table.tsx b/source/frontend/src/components/common/Table/Table.tsx
index 374b933f0..8bc33f34c 100644
--- a/source/frontend/src/components/common/Table/Table.tsx
+++ b/source/frontend/src/components/common/Table/Table.tsx
@@ -565,7 +565,7 @@ export default function Table<RowItem extends Record<string, any>>({
 
   // Handle shift+click range selection
   const handleRowClick = (e: React.MouseEvent, clickedIndex: number) => {
-    if (!table || !table.getRowModel) {
+    if (!table || !table.getRowModel || !tableProps.enableRowSelection) {
       return;
     }
 
@@ -636,7 +636,7 @@ export default function Table<RowItem extends Record<string, any>>({
 
     content = (
       <>
-        <MRT_TopToolbar table={table} />
+        {(tableProps.enableTopToolbar ?? true) && <MRT_TopToolbar table={table} />}
         <MuiTable
           sx={{
             display: 'grid',
