diff --git a/source/app/electron/main.ts b/source/app/electron/main.ts
index a2ac8e598..85a966989 100644
--- a/source/app/electron/main.ts
+++ b/source/app/electron/main.ts
@@ -667,7 +667,6 @@ async function getShell(): Promise<string> {
  */
 async function getShellEnv(): Promise<NodeJS.ProcessEnv> {
   const execPromisify = promisify(exec);
-  const shell = await getShell();
   const isWindows = process.platform === 'win32';
 
   // For Windows, just return the current environment
@@ -675,7 +674,8 @@ async function getShellEnv(): Promise<NodeJS.ProcessEnv> {
     return { ...process.env };
   }
 
-  // For Unix-like systems
+  // For Unix-like systems, get the shell
+  const shell = await getShell();
   const isZsh = shell.includes('zsh');
   // interactive is supported only on zsh
   const shellArgs = isZsh ? ['--login', '--interactive', '-c'] : ['--login', '-c'];
