diff --git a/source/frontend/src/components/project/ProjectDetails.tsx b/source/frontend/src/components/project/ProjectDetails.tsx
index d55a573..a61b69c 100644
--- a/source/frontend/src/components/project/ProjectDetails.tsx
+++ b/source/frontend/src/components/project/ProjectDetails.tsx
@@ -332,7 +332,13 @@ function ProjectOverview({
       </Grid>
 
       {projectSections.map(section => (
-        <Grid key={section.id} item xs={12} md={4}>
+        <Grid
+          key={section.id}
+          item
+          xs={12}
+          md={4}
+          sx={{ '&:has(.MuiCardContent-root:empty)': { display: 'none' } }}
+        >
           <Card sx={{ height: '100%' }}>
             <CardContent>
               <section.component project={project} projectResources={projectResources} />
